es-core/src/PowerSaver.cpp

```diff
@@ -46,6 +46,7 @@ void PowerSaver::updateMode()
 	} else {
 		mMode = DEFAULT;
 	}
+	updateTimeouts();
```

Comment:
Do we need to update Timeouts every time we get Mode?

Also, I see you decided to create an enum that's both the mode and the time for sleep. 

For the sake of readability, for someone who's going to read and maintain the code in the future, is "getMode()" the right name for the method? Should it be "getModeTimeout()" - or should we have an inline "getModeTimeout()" that returns "getMode()" given the current implementation?